### PR TITLE
Add more details to DebugHelper docs

### PR DESCRIPTION
**Pull request by siaw23**
### Summary

I felt the docs needed more information on why Marshal could fail. Also update the `debug` output to match what Rails really puts on the screen.

### CODE REVIEW

File: actionview/lib/action_view/helpers/debug_helper.rb [hunk 1]
```diff
@@ -5,32 +5,71 @@
 module ActionView
   # = Action View Debug Helper
   #
-  # Provides a set of methods for making it easier to debug Rails objects.
+  # Provides a method to debug Rails objects.
```
Comment (simi) [inline actionview/lib/action_view/helpers/debug_helper.rb:8]: If we're going to touch this line, we can move from `Rails` to `Ruby` here. There is nothing such a `Rails object` AFAIK.

**Comment by simi** [comment]
🤔 This seems too verbose to me. IMHO there is no need to provide whole list of possible failures.

The main mystery still remains: Do we need to keep `Marshal.dump(object)`line today?

File: actionview/lib/action_view/helpers/debug_helper.rb (hunk 1 above)
Comment (siaw23) [reply actionview/lib/action_view/helpers/debug_helper.rb]: I removed "Rails" entirely, if we're in Rails I think we can't debug any other objects than Ruby objects. wdyt?

**Comment by siaw23** [comment]
> The main mystery still remains: Do we need to keep `Marshal.dump(object)`line today?

The main reason that I touch this file TBH is I looked at `Marshal.dump` for a long time wondering why it's there without any explanation. I removed `Marshal.dump`. Not sure what others will make of it. 

/cc @byroot @jonathanhefner

**Comment by byroot** [comment]
I usually don't have much opinion on docs as I'm terrible at it.

From a maintainer standpoint however I think the existing comment is enough to explain the intent.

File: actionview/lib/action_view/helpers/debug_helper.rb [hunk 2]
```diff
@@ -5,32 +5,64 @@
...
+      #
       #   # =>
-      #   <pre class='debug_dump'>--- !ruby/object:User
-      #   attributes:
-      #     updated_at:
-      #     username: testing
-      #     age: 42
-      #     password: xyz
-      #     created_at:
+      #   <pre class="debug_dump">--- !ruby/object:User
```
Comment (simi) [inline actionview/lib/action_view/helpers/debug_helper.rb:21]: This seems too verbose to me. Original example was simple enough to get clear picture what's happening in 8 lines.

Comment (siaw23) [reply actionview/lib/action_view/helpers/debug_helper.rb:21]: Reasonable. I agree.

**Comment by simi** [comment]
Also as I mentioned earlier, there is this test which relies on the `Marshal.dump` presence -> https://github.com/rails/rails/blob/3bcd48f6d8b0d87e02b75d124984895cf28e6f9c/actionview/test/activerecord/debug_helper_test.rb#L15-L18.

```ruby
irb(main):008:0> Marshal.dump( -> {})
(irb):8:in `dump': no _dump_data is defined for class Proc (TypeError)
        from (irb):8:in `<main>'                                   
        from /home/retro/.rubies/ruby-3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /home/retro/.rubies/ruby-3.1.2/bin/irb:25:in `load'   
        from /home/retro/.rubies/ruby-3.1.2/bin/irb:25:in `<main>' 
irb(main):009:0> (-> {}).to_yaml
=> "--- !ruby/object:Proc {}\n"
```

Personally I have no idea what could be the impact of removing `Marshal.dump` (and related test).

---

:information_source: @siaw23 not sure how [ci-skip] is supported on buildkite, but since you're touching code (not documentation only), you should not use it in this case.

**Comment by siaw23** [comment]
OK, let's kill this PR. :)

**Pull request closed without merging**